### PR TITLE
Guard against document not being defined

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

**Comment by defunctzombie** [comment]
Why is document not defined? This module expects document to be defined.

**Comment by KyleAMathews** [comment]
When rendering React.js apps on the server.

**Comment by defunctzombie** [comment]
What is the intended behavior? This PR just makes this module a no-op?

Also, the diff is very noisy, would be better to do the if not document return at the top and leave the rest of the code as is.

**Comment by KyleAMathews** [comment]
Yes, the intended behavior is to make this module a noop in non-browser environments.

Agree much better to have simple return at the top. Updated commit.

File: index.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
+// Guard against document not being defined in non-browser environments.
+if (typeof document !== 'undefined') {
+  module.exports = function (){};
```
Comment (defunctzombie) [inline index.js]: 4 spaces to be consistent with file

File: index.js [hunk 2]
```diff
@@ -1,3 +1,8 @@
+// Guard against document not being defined in non-browser environments.
+if (typeof document !== 'undefined') {
```
Comment (defunctzombie) [inline index.js]: should this not be ` typeof global.document` ? Since we always access via global variable

**Comment by KyleAMathews** [comment]
Fixed.

File: index.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
+// Guard against document not being defined in non-browser environments.
+if (typeof global.document !== 'undefined') {
```
Comment (defunctzombie) [inline index.js]: Totally missed this but this if check is backwards.. I have fixed it in master. Please test PRs :) My oversight in not testing this better before merging.

Comment (KyleAMathews) [inline index.js]: Oops! I was getting there :)

**Pull request merged**